lib/LanguageServerRename/Model/Renamer/MemberRenamer.php

```diff
@@ -0,0 +1,182 @@
+<?php
+
+namespace Phpactor\Extension\LanguageServerRename\Model\Renamer;
+
+use Generator;
+use Microsoft\PhpParser\Node;
+use Microsoft\PhpParser\Node\ClassConstDeclaration;
+use Microsoft\PhpParser\Node\ConstElement;
+use Microsoft\PhpParser\Node\Expression\AssignmentExpression;
+use Microsoft\PhpParser\Node\Expression\MemberAccessExpression;
+use Microsoft\PhpParser\Node\Expression\ScopedPropertyAccessExpression;
+use Microsoft\PhpParser\Node\Expression\Variable;
+use Microsoft\PhpParser\Node\MethodDeclaration;
+use Microsoft\PhpParser\Node\PropertyDeclaration;
+use Microsoft\PhpParser\Parser;
+use Microsoft\PhpParser\Token;
+use Phpactor\Extension\LanguageServerRename\Model\NodeUtils;
+use Phpactor\Extension\LanguageServerRename\Model\RenameLocationGroup;
+use Phpactor\Extension\LanguageServerRename\Model\RenameLocationsProvider;
+use Phpactor\Extension\LanguageServerRename\Model\RenameResult;
+use Phpactor\Extension\LanguageServerRename\Model\Renamer;
+use Phpactor\TextDocument\ByteOffset;
+use Phpactor\TextDocument\ByteOffsetRange;
+use Phpactor\TextDocument\Location;
+use Phpactor\TextDocument\TextDocument;
+use Phpactor\TextDocument\TextDocumentLocator;
+use Phpactor\TextDocument\TextDocumentUri;
+use Phpactor\TextDocument\TextEdit;
+use Phpactor\TextDocument\TextEdits;
+
+class MemberRenamer implements Renamer
+{
+    /** @var Parser */
+    private $parser;
+    /** @var NodeUtils */
+    private $nodeUtils;
+    /** @var TextDocumentLocator */
+    private $locator;
+    /** @var RenameLocationsProvider */
+    private $locationsProvider;
+
+    public function __construct(
+        Parser $parser,
+        NodeUtils $nodeUtils,
+        TextDocumentLocator $locator,
+        RenameLocationsProvider $locationsProvider
+    ) {
+        $this->parser = $parser;
+        $this->nodeUtils = $nodeUtils;
+        $this->locator = $locator;
+        $this->locationsProvider = $locationsProvider;
+    }
+
+    private function getValidNode(TextDocument $textDocument, ByteOffset $offset): ?Node
```

Comment:
order methods `public`, `protected` and tinally `private`. this helps promote the `public` API